### ISSUE TITLE
Fatal crash in GreedyTokenSampler.sampleFromProbs when top-K probability sum is zero

### ISSUE THREAD

**Issue by tsay-dev** [argmaxinc/WhisperKit#478]
## Summary

`GreedyTokenSampler.sampleFromProbs` crashes with `Fatal error: Can't get random value with an empty range` when the sum of top-K probabilities is zero.

## Location

`Sources/TTSKit/Utilities/Sampling.swift`, line 239 (v1.0.0):

```swift
let probSum = probsArray.reduce(0, +)
let randomValue = Float.random(in: 0..<probSum, using: &rng)  // crashes when probSum == 0``

## Cause
When softmax outputs become numerically unstable (e.g., all logits are -infinity or extremely small), the top-K probabilities all round to 0. This makes prob​Sum = 0, resulting in Float​.random(in: 0..<0) — an empty range that triggers a Swift runtime fatal​Error.

## Suggested Fix
Guard against prob​Sum <= 0 and fall back to the highest-probability token (argmax):

```swift
guard probSum > 0 else {
    let maxIdx = probsArray.enumerated().max(by: { $0.element < $1.element })?.offset ?? 0
    return Int32(idxArray[maxIdx])
}
let randomValue = Float.random(in: 0..<probSum, using: &rng)``

## Environment
• TTSKit v1.0.0
• iOS 18+
• Observed during on-device TTS generation

**Comment by sumitvairagar** [comment]
Submitted a fix in PR #531.

The crash is in `sampleFromProbs` (still present in v1.1.0 despite the refactor) — the `Float.random(in: 0..<probSum)` call when `probSum == 0`. Added a `guard probSum > 0` with an argmax fallback, and a unit test that reproduces the condition (all logits set to `-Float.infinity` except one) to prevent regression.
